klog.go

```diff
@@ -952,18 +965,30 @@ const bufferSize = 256 * 1024
 // l.mu is held.
 func (l *loggingT) createFiles(sev severity) error {
 	now := time.Now()
-	// Files are created in decreasing severity order, so as soon as we find one
-	// has already been created, we can stop.
-	for s := sev; s >= infoLog && l.file[s] == nil; s-- {
+	if l.logFile != "" { // single-file mode
 		sb := &syncBuffer{
 			logger:   l,
-			sev:      s,
+			sev:      infoLog,
 			maxbytes: CalculateMaxSize(),
 		}
 		if err := sb.rotateFile(now, true); err != nil {
```

Comment:
This is sort of weird, since it's not actually rotating the files in single-file mode. I guess it's fine for now though.